packages/ics721/src/execute.rs

```diff
@@ -258,8 +260,28 @@ where
         let instantiate = if CLASS_ID_TO_NFT_CONTRACT.has(deps.storage, class.id.clone()) {
             vec![]
         } else {
+            let class_id = ClassId::new(class.id.clone());
+            // for creating a predictable nft contract using, using instantiate2, we need: checksum, creator, and salt:
+            // - using class id as salt for instantiating nft contract guarantees a) predictable address and b) uniqueness
+            // for this salt must be of length 32 bytes, so we use sha256 to hash class id
+            let mut hasher = Sha256::new();
+            hasher.update(class_id.as_bytes());
+            let salt = hasher.finalize().to_vec();
+            // Get the canonical address of the contract creator
+            let canonical_creator = deps.api.addr_canonicalize(env.contract.address.as_str())?;
+            // get the checksum of the contract we're going to instantiate
+            let CodeInfoResponse { checksum, .. } = deps
+                .querier
+                .query_wasm_code_info(CW721_CODE_ID.load(deps.storage)?)?;
+            let canonical_cw721_addr = instantiate2_address(&checksum, &canonical_creator, &salt)?;
```

Comment:
Can get a quick explanation of what is going on here?

Reply:
This is mainly explained in docs here: https://github.com/CosmWasm/cosmwasm/blob/main/packages/std/src/addresses.rs#L270-L309

Essentially it uses salt for creating a unique address. Since `class_id` is unique for ics721, it is an ideal candidate for salt. NFT module has a exactly same approach: instantiated nft contract is a hash of class id.